### PR TITLE
Kim/wnyc 310

### PR DESCRIPTION
**Pull request by kimlarocca**

### CODE REVIEW

File: components/SUser.vue [hunk 1]
```diff
@@ -53,22 +52,7 @@ const onLogIn = () => {
...
-  // set the currentUser composable to null
-  currentUser.value = null
-
-  // clear what is playing
-  currentEpisode.value = null
-  currentEpisodeHolder.value = null
-  isEpisodePlaying.value = false
-
-  getAndSetUserProfile()
+  logOutUser()
```
Comment (bonomite) [inline components/SUser.vue]: wrapped the logout code in a helper function

File: pages/delete.vue [hunk 2]
```diff
@@ -1,15 +1,18 @@
 <script setup>
 import { requestAccountDeletion } from "~/utilities/helpers"
 import { useCurrentUserProfile } from "~/composables/states.ts"
-import VFlexibleLink from '@nypublicradio/nypr-design-system-vue3/v2/src/components/VFlexibleLink.vue'
+import VFlexibleLink from "@nypublicradio/nypr-design-system-vue3/v2/src/components/VFlexibleLink.vue"
 const currentUserProfile = useCurrentUserProfile()
 </script>
 
 <template>
+  <div>
```
Comment (bonomite) [inline pages/delete.vue:9]: needed to add this div or the transitions were messed up

File: pages/delete.vue [hunk 3]
```diff
@@ -36,19 +39,31 @@ const currentUserProfile = useCurrentUserProfile()
...
-        />
+        <div class="s-user my-8">
+          <Avatar
+            v-if="currentUserProfile?.avatar_image_url"
+            :image="currentUserProfile?.avatar_image_url"
+            size="large"
+            shape="circle"
+          />
+          <div v-else class="p-avatar">
+            <UserIcon />
```
Comment (bonomite) [inline pages/delete.vue:55]: if no image exists, add the fallback user icon component

File: utilities/helpers.ts [hunk 4]
```diff
@@ -1135,14 +1124,30 @@ export const requestAccountDeletion = async () => {
...
   // send user to the sign in page
   await navigateTo('/')
-  
+
   // show toast confirmation of deletion request
   globalToast.value = {
     severity: "info",
     summary: 'We have received your request to delete your account. Please allow 7-10 business days for your request to be processed.',
-    life: 3000,
+    closable: true,
```
Comment (bonomite) [inline utilities/helpers.ts:1152]: removed the life timer so the user can read what it says and close it themselves

File: utilities/helpers.ts [hunk 5]
```diff
@@ -1135,14 +1124,30 @@ export const requestAccountDeletion = async () => {
...
+
+  // post to zapier webhook
+  if (currentUserProfile.value?.id) {
+    await $fetch('https://hooks.zapier.com/hooks/catch/1135793/23fbxa5/', {
+      method: 'POST',
+      body: { "email": currentUserProfile.value?.email, "id": currentUserProfile.value?.id }
+    });
+  }
+
+  logOutUser()
```
Comment (bonomite) [inline utilities/helpers.ts:1143]: using the new logOutUser helper function here

File: utilities/helpers.ts [hunk 6]
```diff
@@ -1105,24 +1105,13 @@ export const toggleAskNotificationPermisstions = async (isEnabled = true) => {
   }
 }
 
-// handle account deletion requests
-export const requestAccountDeletion = async () => {
+export const logOutUser = async () => {
```
Comment (bonomite) [inline utilities/helpers.ts:1109]: new logOutUser helper function

File: components/CardLarge.vue [hunk 7]
```diff
@@ -47,7 +47,7 @@ defineExpose({
         <HtmlConvert :htmlContent="props.item.tease" class="desc" />
 
         <PipeData class="text-xs" :hidePipe="props.hideDate">
-          <template #left>{{ props.item.headers.brand.title }}</template>
+          <template #left>{{ props.item?.headers?.brand.title }}</template>
```
Comment (kimlarocca) [inline components/CardLarge.vue:50]: good idea

File: components/SUser.vue (hunk 1 above)
Comment (kimlarocca) [reply components/SUser.vue]: much better!

**Review by kimlarocca — COMMENTED** [review]
looks great thank you!

**Review by bonomite — APPROVED** [review]

**Pull request merged**